react/features/calendar-sync/middleware.js

```diff
@@ -34,34 +45,77 @@ MiddlewareRegistry.register(store => next => action => {
     return result;
 });
 
+/**
+ * Clears the calendar access status when the app comes back from
+ * the background. This is needed as some users may never quit the
+ * app, but puts it into the background and we need to try to request
+ * for a permission as often as possible, but not annoyingly often.
+ *
+ * @private
+ * @param {Object} store - The redux store.
+ * @param {Object} action - The Redux action.
+ * @returns {void}
+ */
+function _maybeClearAccessStatus(store, action) {
+    const { appState } = action;
+
+    if (appState === 'background') {
+        const { dispatch } = store;
+
+        dispatch(updateCalendarAccessStatus(undefined));
+    }
+}
+
 /**
  * Ensures calendar access if possible and resolves the promise if it's granted.
  *
  * @private
  * @param {boolean} promptForPermission - Flag to tell the app if it should
  * prompt for a calendar permission if it wasn't granted yet.
+ * @param {Function} dispatch - The Redux dispatch function.
  * @returns {Promise}
  */
-function _ensureCalendarAccess(promptForPermission) {
+function _ensureCalendarAccess(promptForPermission, dispatch) {
     return new Promise((resolve, reject) => {
         RNCalendarEvents.authorizationStatus()
             .then(status => {
                 if (status === 'authorized') {
-                    resolve();
+                    resolve(true);
                 } else if (promptForPermission) {
-                    RNCalendarEvents.authorizeEventStore()
-                        .then(result => {
-                            if (result === 'authorized') {
-                                resolve();
-                            } else {
-                                reject(result);
-                            }
-                        })
-                        .catch(error => {
-                            reject(error);
-                        });
+                    if (Platform.OS === 'ios' && status === 'denied') {
+                        /**
+                         * This is an iOS device where we cannot ask for a
+                         * permission again if the user has already
+                         * denied it earler.
+                         */
+                        Alert.alert(
+                             'Permission required',
+                             // eslint-disable-next-line max-len
+                             'Calendar permission is required to list your meetings in the app.',
+                             [
+                                 { text: 'Cancel' },
+                                 {
+                                     onPress: openSettings,
+                                     text: 'Settings'
+                                 }
+                             ],
+                             { cancelable: false });
+
+                        dispatch(updateCalendarAccessStatus(status));
```

Comment:
Is it okay that resolve/reject are not called in the ios/denied block?

Reply:
Firstly, `Alert` is not declarative but imperative so it's not our buddy and we try to stay away from it if possible. More importantly, I'm afraid of multiple `Alert`s (possibly when we get to automatically trying to refresh and rerequest). I think I'd rather have the message with the link to the system 'Settings' in the center of the 'Calendar' tab's content. Because there's no permission, nothing will be displayed in the 'Calendar' tab's content and we may as well fill it with what's now in the alert.

Reply:
Shit Lenny, you're right, I don't know how it worked properly this way... actually the if in the .then branch also wasn't properly written, fixed both.

Lyubo: actually I brought this piece of code from the video permission part (features/mobile/permission), but I understand your concerns and change this to a permanent component